include/aspect/particle/property/interface.h

```diff
@@ -623,6 +633,11 @@ namespace aspect
            */
           ParticlePropertyInformation property_information;
 
+          /**
+           * This object owns and organizes the memory for all particle
+           * properties.
+           */
+          std_cxx11::unique_ptr<PropertyPool> property_pool;
```

Comment:
Do you make this a pointer instead of a regular member variable because you don't know the number of property components yet at the time you construct the enclosing object? If so, maybe add this information to the comment.